src/dmd/typesem.d

```diff
@@ -3619,7 +3619,7 @@ Expression dotExp(Type mt, Scope* sc, Expression e, Identifier ident, int flag)
                 e.error("`%s` is not an expression", e.toChars());
                 return ErrorExp.get();
             }
-            else if (checkUnsafeDotExp(sc, e, ident, flag))
+            else if (mt.dim.toUInteger() < 1 && checkUnsafeDotExp(sc, e, ident, flag))
```

Comment:
A comment explaining why this clause is here would be good

Reply:
A comment with a link to the bugzilla issue should suffice.

Reply:
Done